File: Estágio AutoMind/Usuario/CadastroUsuario.cs
namespace EstagioAutomind.Usuario;

public class CadastroUsuario
{
	/* Classe de cadastro de usuário com parâmetros e encapsulamento
		get para retorno e visualização e set para modificação */
	public string Nome {get; set;}
	public string Email {get; set;}
	public int  Idade {get; set;}
	
	/* Construtores com argumentos */
	public CadastroUsuario(string nome, string email, int idade) {
		Nome = nome;
		Email = email;
		Idade = idade;
	}
	
	public static void Cadastro(List<CadastroUsuario> usuario) {
		int temp = -1; /* variável para controle do loop doWhile */
		do {
			Console.WriteLine("\n*** Cadastro Usuário ***");
			Console.Write("Nome: ");
			string nome = Console.ReadLine();
			Console.Write("Email: ");
			string email = Console.ReadLine();
			
			/* Tratamento de entrada para idade */
			int idade;
			bool idadeValida = false; /* booleano para validação */
			while (!idadeValida) {
				Console.Write("Idade: ");
				string IdadeEntrada = Console.ReadLine();
				
				if (int.TryParse(IdadeEntrada, out idade) && idade > 0) {
					idade = int.Parse(IdadeEntrada);
					idadeValida = true;
					
					/* returna instância de cadastro */
					CadastroUsuario DadoUsuario = new CadastroUsuario(nome, email, idade);
					usuario.Add(DadoUsuario);
				} else {
					Console.WriteLine("Por favor, insira uma idade válida (um número inteiro positivo).");
				}
			}
			
			/* Tratamento para continuação de cadastro */ 
			bool continuar = false;
			
			while (!continuar) {
				Console.Write("\nDeseja cadastrar mais um usuário? [1 sim | 0 não]: ");
				string continuarCadastro = Console.ReadLine();
				
				if (int.TryParse(continuarCadastro, out temp)) {
					/* se for um inteiro, validar a opção */
					if (temp == 0 || temp == 1) {
						temp = int.Parse(continuarCadastro);
						continuar = true;
					} else { Console.WriteLine("Opção inválida. Por favor, insira '1' para sim ou '0' para não.");}
				} else { Console.WriteLine("Só é permitido inteiro"); }
			}
		} while (temp == 1);
	}
}

File: Estágio AutoMind [Java]/Automind_java/src/Usuario/CadastroUsuario.java
package Usuario;

import java.util.List;
import java.util.Scanner;

public class CadastroUsuario {
    public String nome;
    public String email;
    public Integer idade;

    public CadastroUsuario(String nome, String email, Integer idade) {
        this.nome = nome;
        this.email = email;
        this.idade = idade;
    }

    public String getNome() {
        return nome;
    }

    public String getEmail() {
        return email;
    }

    public Integer getIdade() {
        return idade;
    }

    public static void cadastro(List<CadastroUsuario> usuarios, Scanner sc){
        int opcao = -1, entradaOpcao;
        int idade = -1, entradaIdade;



        do {
            boolean validador = false;
            boolean validador_2 = false;

            System.out.println();
            System.out.println("*** Seção cadastro ***");
            System.out.print("Insira o nome do usuário: ");
            String nome = sc.nextLine();
            System.out.print("Insira o email do usuário: ");
            String email = sc.nextLine();

            while (!validador) {
                System.out.print("Insira a idade: ");
                String entrada = sc.nextLine();

                try {
                    entradaIdade = Integer.parseInt(entrada);
                    if (entradaIdade > 0){
                        validador = true;
                        idade = entradaIdade;

                        CadastroUsuario dadoUsuario = new CadastroUsuario(nome, email, idade);
                        usuarios.add(dadoUsuario);

                    } else { System.out.println("Insira uma idade válida!"); }
                } catch (NumberFormatException e) {
                    System.out.println("Só é permitido número inteiro");
                }
            }

            while (!validador_2) {
                System.out.print("\nDeseja cadastrar um novo usuário? [1 sim | 2 não]: ");
                String entrada_2 = sc.nextLine();

                try {
                    entradaOpcao = Integer.parseInt(entrada_2);
                    if (entradaOpcao >= 1 && entradaOpcao <= 2) {
                        validador_2 = true;
                        opcao = entradaOpcao;
                    } else { System.out.println("\nErro! Insira 1 para sim e 2 para não."); }
                } catch (NumberFormatException e) {
                    System.out.println("\nErro! só é permitido inteiro - Insira 1 para sim e 2 para não.");
                }
            }
        } while (opcao == 1);

    }
}
